halo2_proofs/src/dev.rs

```diff
@@ -746,8 +746,26 @@ impl<F: FieldExt> MockProver<F> {
     /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
     /// the reasons that the circuit is not satisfied.
     pub fn verify(&self) -> Result<(), Vec<VerifyFailure>> {
+        self.verify_at_rows(self.usable_rows.clone())
```

Comment:
Can we leave `verify` with the regular behavior and leave it co-existing with `verify_at_rows`?

Reply:
in fact the real logic of current `verify` is identical to old `verify`. By changing it to `self.verify_at_rows(self.usable_rows.clone())` we can avoid keeping 2 copy of hundreds of lines of very similar codes

Reply:
hmmm. kk sure. Since it only affects the `MockProver` it should not be an issue.